torch/_inductor/codegen/cpp_template_kernel.py

```diff
@@ -0,0 +1,204 @@
+import itertools
+from typing import Any, Callable, Dict, List, Optional, Tuple, Union
+
+import sympy
+from sympy.parsing.sympy_parser import parse_expr
+
+import torch
+
+from torch._inductor.autotune_process import CppBenchmarkRequest
+from torch._inductor.utils import sympy_index_symbol
+from .. import codecache, config, ir, lowering as L
+from ..virtualized import V
+from .common import Kernel, OpOverrides
+from .cpp_utils import cexpr_index, DTYPE_TO_CPP
+
+
+def parse_expr_with_index_symbols(expr_str: str) -> sympy.Expr:
+    expr = parse_expr(expr_str)
+    int_symbols = {sym: sympy_index_symbol(sym.name) for sym in expr.free_symbols}
+    return expr.subs(int_symbols)
+
+
+def wrap_with_tensorbox(node) -> ir.TensorBox:
+    return (
+        ir.TensorBox.create(node) if isinstance(node, ir.Buffer) else ir.TensorBox(node)
+    )
+
+
+class CppTemplateKernel(Kernel):
+    overrides = OpOverrides
+
+    def __init__(self, kernel_name):
+        super().__init__()
+        self.kernel_name = kernel_name
+
+    def def_kernel(
+        self,
+        inputs: List[Tuple[str, ir.Buffer]],
+        outputs: List[Tuple[str, ir.Buffer]],
```

Comment:
dicts are ordered in python, so you can use dicts here

Reply:
Thanks. Fixed.